lib/facter/util/resolvers/networking/networking.rb

```diff
@@ -36,7 +36,8 @@ def expand_main_bindings(networking_facts)
             interfaces = networking_facts[:interfaces]
 
             expand_interfaces(interfaces) unless interfaces.nil?
-            expand_primary_interface(networking_facts, primary) unless primary.nil? || interfaces.nil?
+            expand_primary_interface(networking_facts, primary) unless
+              primary.nil? || interfaces.nil? || networking_facts.nil?
```

Comment:
To me, ending a statement with `unless` makes sense when it's brief; once it's long enough to be broken up with a line break, it seems more readable to use the full syntax:
```
unless primary.nil? || interfaces.nil? || networking_facts.nil?
  expand_primary_interface(networking_facts, primary)
end
```

Reply:
Oh I didn't know that was the full expression!

Reply:
Hmm...Rubocop isn't happy if there isn't also an else after

Reply:
Ah Rubocop suggested to use a guard clause instead so my change will be a lil different but still same idea!